feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/CameraXViewfinder.kt

```diff
@@ -121,6 +133,25 @@ fun CameraXViewfinder(
             }
     }
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        LaunchedEffect(Unit) {
+            snapshotFlow { viewfinderArgs }
+                .filterNotNull()
+                .map { args ->
+                    if (args.isSourceHdr &&
+                        args.implementationMode == ImplementationMode.PERFORMANCE
+                    ) {
+                        ActivityInfo.COLOR_MODE_HDR
+                    } else {
+                        ActivityInfo.COLOR_MODE_DEFAULT
+                    }
+                }.distinctUntilChanged()
+                .onEach { currentOnRequestWindowColorMode(it) }
+                .onCompletion { currentOnRequestWindowColorMode(ActivityInfo.COLOR_MODE_DEFAULT) }
```

Comment:
What about just replacing this with:
```
.onEach { configuration.colorMode = it }
.onCompletion { configuration.colorMode = ActivityInfo.COLOR_MODE_DEFAULT }
```
Where you define configuration as:
```
val configuration = LocalConfiguration.current
LaunchedEffect(configuration) {
  ..
}
```

Reply:
I'm not quite following how this would work. Are you saying the configuration would be defined at a higher level, such as `JcaApp`? 

I'm not seeing much documentation on `LocalConfiguration`, but from the documentation I've seen for `CompositionLocal`, it doesn't seem to mention using `CompositionLocal` to communicate upwards. In fact, it feels like the current code follows the alternative of [inversion of control](https://developer.android.com/develop/ui/compose/compositionlocal#inversion-of-control).

Is this a common thing to do with `LocalConfiguration`?